distribution/tools/launchers/src/main/java/org/elasticsearch/tools/launchers/JvmOptionsParser.java

```diff
@@ -53,82 +55,88 @@
      * The main entry point. The exit code is 0 if the JVM options were successfully parsed, otherwise the exit code is 1. If an improperly
      * formatted line is discovered, the line is output to standard error.
      *
-     * @param args the args to the program which should consist of a single option, the path to the JVM options
+     * @param args the args to the program which should consist of a single option, the path to ES_PATH_CONF
      */
     public static void main(final String[] args) throws InterruptedException, IOException {
         if (args.length != 1) {
-            throw new IllegalArgumentException("expected one argument specifying path to jvm.options but was " + Arrays.toString(args));
+            throw new IllegalArgumentException("expected one argument specifying path to ES_PATH_CONF but was " + Arrays.toString(args));
         }
+
+        final Stream<Path> jvmOptionsFiles;
+        final Path rootJvmOptions = Paths.get(args[0], "jvm.options");
+        final Path jvmOptionsDirectory = Paths.get(args[0], "jvm.options.d");
+        if (Files.isDirectory(jvmOptionsDirectory)) {
+            final Stream<Path> jvmOptionsDirectoryFiles = Files.list(Paths.get(args[0], "jvm.options.d"));
```

Comment:
Should we files the files e.g. only include those ending with `.options`? Then we'd filter out any backup files, text files etc.

Reply:
I incorporated this idea.